### PR TITLE
Issue 478 maven plugin

### PR DESCRIPTION
**Pull request by l-1squared**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -258,86 +237,80 @@ configure(subprojects) {
...
+        doFirst {
+            if (!isCI && isReleaseVersion && !JavaVersion.current().isJava8Compatible()) {
+                throw new GradleException("Releases have to be done with Java 8")
+            }
+        }
+    }
+    publishing {
+        publications {
+            maven(MavenPublication) {
+                from components.java
```
Comment (l-1squared) [inline build.gradle:253]: This currently fails the `jgiven-android` build :(

Comment (l-1squared) [reply build.gradle:219]: not really, because I duplicated everything

File: scripts/release.sh [hunk 2]
```diff
@@ -33,7 +37,7 @@ if [ -n "$(git status --porcelain)" ]; then
 fi
 
 echo Building, Testing, and Uploading Archives...
-./gradlew --no-parallel clean test install uploadArchives
+./gradlew --no-parallel clean test publishToMaven -Pstaging
```
Comment (NilsOliverLinden) [inline scripts/release.sh:40]: Currently, I do not see why you need the staging profile here. In the project there is no staging profile defined, is it?
Only in the gradle-example projects exist a staging profile to determine from where the plugin should be loaded.

File: build.gradle [hunk 3]
```diff
@@ -10,13 +10,11 @@ plugins {
 apply plugin: 'project-report'
 
 description = 'JGiven - BDD in plain Java'
-def jacocoEnabled = !JavaVersion.current().isJava11Compatible();
-
-
+def jacocoEnabled = !JavaVersion.current().isJava11Compatible()
 
 wrapper {
```
Comment (aaschmid) [inline build.gradle:15]: `wrapper` section should not be needed and is not part of at least my repository anymore, see e.g. https://github.com/aaschmid/gradle-cpd-plugin/blob/master/build.gradle

File: build.gradle [hunk 4]
```diff
@@ -43,6 +41,7 @@ allprojects {
 }
 
 
+
```
Comment (aaschmid) [inline build.gradle:44]: Optional: get rid of too many newlines ;-)

File: build.gradle [hunk 5]
```diff
@@ -78,18 +77,81 @@ subprojects {
...
         hsqldbVersion = '2.3.2'
         byteBuddyVersion = '1.10.2'
-        androidGradleVersion= '3.1.0'
-        androidCompileSdkVersion= 27
-        androidBuildToolsVersion= '27.0.2'
+        androidGradleVersion = '3.1.0'
+        androidCompileSdkVersion = 27
+        androidBuildToolsVersion = '27.0.2'
+
+        publishVariant = {String variant->
```
Comment (aaschmid) [inline build.gradle:84]: Maybe add spaces -> `{ String variant ->`

File: build.gradle [hunk 6]
```diff
@@ -221,13 +265,14 @@ configure(subprojects.findAll { !it.name.contains("android") }) {
     }
 
     copyAsciiDoc.finalizedBy(asciidoctor)
+
+    project.ext.publishVariant('java')
```
Comment (aaschmid) [inline build.gradle:269]: I still find it a bit confusing to use `ext` here. 
But haven't a better idea now.

File: scripts/release.sh [hunk 7]
```diff
@@ -18,6 +18,10 @@ if [[ ! $1 =~ ^[0-9]*\.[0-9]*\.[0-9]*(-[A-Z0-9]*)?$ ]]; then
     exit 1
 fi
 
+if [ -z $ANDROID_SDK_ROOT ]; then
+    echo "Android package will NOT be released"
```
Comment (aaschmid) [inline scripts/release.sh:22]: Shouldn't we abort rather abort the release as you want to normally release it together?

File: settings.gradle [hunk 8]
```diff
@@ -52,16 +52,17 @@ include ':jgiven-core',
...
 def release = System.env.RELEASE == "true"
 def android = System.env.ANDROID == "true"
+boolean hasAndroidSourcePath = ( System.env.ANDROID_SDK_ROOT )
 
 if (JavaVersion.current().isJava9Compatible()) {
     include ':jgiven-java9-tests'
 }
 
-if (android || release) {
+if ((android || release) && hasAndroidSourcePath) {
```
Comment (aaschmid) [inline settings.gradle:61]: Is this excluded for developers which have no Android SDK?

Comment (l-1squared) [reply settings.gradle:61]: err yes. After all if you attempt to build everything but don't have that SDK path set, then the android build will throw a beatiful error

File: scripts/release.sh (hunk 7 above)
Comment (l-1squared) [reply scripts/release.sh:22]: yeah, that might be better

File: build.gradle (hunk 3 above)
Comment (l-1squared) [reply build.gradle:15]: giving it some thought, I guess you're right. After all this section only replicates the information that we have already checked in in the `gradle-wrapper.properties` file.

File: scripts/release.sh (hunk 2 above)
Comment (l-1squared) [reply scripts/release.sh:40]: yes. Doing that rebase that late in the day probably wasn't a good idea

**Pull request merged**